src/setContextLink.ts

```diff
@@ -0,0 +1,29 @@
+import {
+  Operation,
+  NextLink,
+  FetchResult,
+} from './types';
+import * as Observable from 'zen-observable';
+
+import {
+  ApolloLink,
+} from './link';
+
+export default class SetContextLink extends ApolloLink {
+
+  constructor (private context?: any) {
+    super();
+  }
+
+  public request(operation: Operation, forward: NextLink): Observable<FetchResult> {
+    if (!operation.context) {
+      operation.context = {};
+    }
+    operation.context = {
```

Comment:
This should be a recursive merge, since a link could overwrite an entire object, such as headers